src/plugins/irc-events/link.js

```diff
@@ -93,11 +93,12 @@ function fetch(url, cb) {
 		return;
 	}
 	var length = 0;
-	var limit = 1024 * 10;
+	var limit = Helper.config.prefetchMaxImageSize * 1024;
 	req
 		.on("response", function(res) {
-			if (!(/(text\/html|application\/json)/.test(res.headers["content-type"]))) {
-				res.req.abort();
+			if (!(/(image\/.+)/.test(res.headers["content-type"]))) {
+				// if not image, limit download to 10kb, since we need only meta tags
```

Comment:
This might have to change if we end up proxying videos, etc, which we might end up doing. But for now this looks good.